src/spyglass/utils/dj_mixin.py

```diff
@@ -530,8 +565,138 @@ def delete(self, force_permission=False, *args, **kwargs):
+        if warn:
+            logger.warning("!! Bypassing cautious_delete !!")
+            self._log_use(start=time(), super_delete=True)
         super().delete(*args, **kwargs)
+
+    # ------------------------------- Export Log -------------------------------
+
+    @cached_property
+    def _spyglass_version(self):
+        """Get Spyglass version from dj.config."""
+        from spyglass import __version__ as sg_version
+
+        return ".".join(sg_version.split(".")[:3])  # Major.Minor.Patch
+
+    @cached_property
+    def _export_table(self):
+        """Lazy load export selection table."""
+        from spyglass.common.common_usage import ExportSelection
+
+        return ExportSelection()
+
+    @property
+    def export_id(self):
+        """ID of export in progress.
+
+        NOTE: User of an env variable to store export_id may not be thread safe.
+        Exports must be run in sequence, not parallel.
+        """
+
+        return int(environ.get(EXPORT_ENV_VAR, 0))
+
+    @export_id.setter
+    def export_id(self, value):
+        """Set ID of export using `table.export_id = X` notation."""
+        if self.export_id != 0 and self.export_id != value:
+            raise RuntimeError("Export already in progress.")
+        environ[EXPORT_ENV_VAR] = str(value)
+        exit_register(self._export_id_cleanup)  # End export on exit
+
+    @export_id.deleter
+    def export_id(self):
+        """Delete ID of export using `del table.export_id` notation."""
+        self._export_id_cleanup()
+
+    def _export_id_cleanup(self):
+        """Cleanup export ID."""
+        if environ.get(EXPORT_ENV_VAR):
+            del environ[EXPORT_ENV_VAR]
+        exit_unregister(self._export_id_cleanup)  # Remove exit hook
+
+    def _start_export(self, paper_id, analysis_id):
+        """Start export process."""
+        if self.export_id:
+            logger.info(f"Export {self.export_id} in progress. Starting new.")
+            self._stop_export(warn=False)
+
+        self.export_id = self._export_table.insert1_return_pk(
+            dict(
+                paper_id=paper_id,
+                analysis_id=analysis_id,
+                spyglass_version=self._spyglass_version,
+            )
+        )
+
+    def _stop_export(self, warn=True):
+        """End export process."""
+        if not self.export_id and warn:
+            logger.warning("Export not in progress.")
+        del self.export_id
+
+    def _log_fetch(self, *args, **kwargs):
+        """Log fetch for export."""
+        if not self.export_id or self.database == "common_usage":
+            return
+        logger.debug(f"Export {self.export_id}: fetch()   {self.table_name}")
+
+        restr = self.restriction or True
+        limit = kwargs.get("limit")
+        offset = kwargs.get("offset")
+        if limit or offset:
+            restr = super().fetch(  # Use result as restr if limit/offset
+                restr, as_dict=True, limit=limit, offset=offset
```

Comment:
`fetch` doesn't take restrictions as an argument, so causes an error if you get to here.

I also wasn't able to replace this line with:
```python
restr = (
                super().restrict(restr)
            ).fetch(  # Use result as restr if limit/offset
                as_dict=True, limit=limit, offset=offset
            )
``` 
because `restrict` calls `fetch` calls `_log_fetch` which makes a recursion loop.  

May need to make an option in fetch to override the export log status.  then could do something like:
```python
restr = self.fetch(  
                as_dict=True, limit=limit, offset=offset, ignore_log=True
            )
```

Reply:
Note: changing to 
```python
restr = (self & restr).fetch(  # Use result as restr if limit/offset
                as_dict=True,
                limit=limit,
                offset=offset,
                ignore_log=True,
            )
```
Works but causes long restr strings when combined in following line.  See proposed soln for that in next comment

Reply:
Thanks for the suggestion, I've added an ignore log feature